hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/compact/SparkScheduleCompactionActionExecutor.java

```diff
@@ -58,36 +62,98 @@ public SparkScheduleCompactionActionExecutor(HoodieEngineContext context,
-      // exclude files in pending clustering from compaction.
-      fgInPendingCompactionAndClustering.addAll(fileSystemView.getFileGroupsInPendingClustering().map(Pair::getLeft).collect(Collectors.toSet()));
-      return compactor.generateCompactionPlan(context, table, config, instantTime, fgInPendingCompactionAndClustering);
+  public boolean needCompact(CompactType compactType) {
+    boolean compactable;
+    // return deltaCommitsSinceLastCompaction and lastCompactionTs
+    Tuple2<Integer, String> threshold = checkCompact(compactType);
+    int inlineCompactDeltaCommitMax = config.getInlineCompactDeltaCommitMax();
+    int inlineCompactDeltaElapsedTimeMax = config.getInlineCompactDeltaElapsedTimeMax();
+    switch (compactType) {
+      case COMMIT_NUM:
+        compactable = inlineCompactDeltaCommitMax <= threshold._1;
+        LOG.info(String.format("Trigger compaction when commit_num >=%s", inlineCompactDeltaCommitMax));
+        break;
+      case TIME_ELAPSED:
+        compactable = parseToTimestamp(threshold._2) + inlineCompactDeltaElapsedTimeMax <= parseToTimestamp(instantTime);
+        LOG.info(String.format("Trigger compaction when elapsed_time >=%ss", inlineCompactDeltaElapsedTimeMax));
+        break;
+      case NUM_OR_TIME:
+        compactable = inlineCompactDeltaCommitMax <= threshold._1
+            || parseToTimestamp(threshold._2) + inlineCompactDeltaElapsedTimeMax <= parseToTimestamp(instantTime);
+        LOG.info(String.format("Trigger compaction when commit_num >=%s or elapsed_time >=%ss", inlineCompactDeltaCommitMax,
+                inlineCompactDeltaElapsedTimeMax));
+        break;
+      case NUM_AND_TIME:
+        compactable = inlineCompactDeltaCommitMax <= threshold._1
+            && parseToTimestamp(threshold._2) + inlineCompactDeltaElapsedTimeMax <= parseToTimestamp(instantTime);
+        LOG.info(String.format("Trigger compaction when commit_num >=%s and elapsed_time >=%ss", inlineCompactDeltaCommitMax,
+                inlineCompactDeltaElapsedTimeMax));
+        break;
+      default:
+        throw new HoodieCompactionException("Unsupported compact type: " + config.getInlineCompactType());
+    }
 
-    } catch (IOException e) {
-      throw new HoodieCompactionException("Could not schedule compaction " + config.getBasePath(), e);
+    if (compactable) {
+      LOG.info(String.format("Scheduling compaction: %s. Delta commits found: %s times, and last compaction time is %s.",
+              compactType.name(), threshold._1, threshold._2));
+    } else {
+      LOG.info(String.format("Not scheduling compaction as only %s delta commits was found since last compaction %s."
+                      + "Waiting for %s,or %sms elapsed time need since last compaction %s.", threshold._1,
+              threshold._2, config.getInlineCompactDeltaCommitMax(), config.getInlineCompactDeltaElapsedTimeMax(), threshold._2));
     }
+    return compactable;
   }
 
+  public Long parseToTimestamp(String time) {
```

Comment:
`Timestamp` is not clear, it means `second` or `mills` or something else?

Reply:
`parsedToSeconds` ? sry . Do u have any good name?